src/robot_arm.cpp

```diff
@@ -71,13 +119,77 @@ int RobotArm::getSpeed() {
     return speed;
 }
 
-char *RobotArm::stradd(char *dest, const char *src) {
-    size_t i, j;
-    for (i = 0; dest[i] != '\0'; i++) {
+int RobotArm::receiveGcodeResponse(char *response, size_t responseSize, unsigned int readTimeout) {
+    bool receivingData = true;
+    unsigned int responseCharCounter = 0;
+    char byteRead = 0;
+
+    /// Convert to microseconds
+    readTimeout *= 1000;
+
+    /// Decrease the response size as we will include a \0 character by ourselves.
+    if (responseSize > 0) {
+        responseSize--;
+    }
+
+    unsigned int lastRead = hwlib::now_us();
+
+    while (receivingData) {
+        if (uartConn.available() > 0) {
+            byteRead = uartConn.receive();
+
+            /// Read until we found an endline character.
+            /// If the responseCharCounter does equal the size of the response array, we stop to prevent writing out of memory.
+            if (byteRead != '\n' && responseCharCounter < responseSize) {
+                if (response) {
+                    /// Only if the response is not a null pointer, we write to the response array.
+                    response[responseCharCounter] = byteRead;
+                }
+                responseCharCounter += 1;
+
+                lastRead = hwlib::now_us();
+            } else if (responseCharCounter > 0) { /// We have found a endline. If the response char counter is larger then zero
+                                                  /// (there is data), we will stopping polling for new data.
+                receivingData = false;
+            }
+        }
+
+        /// Check if a timeout occurred. If so, return zero.
+        if ((hwlib::now_us() - lastRead) > readTimeout) {
+            return 0;
+        }
+    }
+
+    /// Add a \0 character to the response, if the is a response.
+    if (responseCharCounter > 0) {
+        response[responseCharCounter++] = '\0';
+    }
+
+    /// Return the amount of characters read
+    return responseCharCounter;
+}
+
+bool RobotArm::isConnected() {
+    uartConn << "#n P2203\n";
+
+    /// By giving a null pointer as a method parameter, we save unnecessarily memory space.
+    if (!receiveGcodeResponse(nullptr, 255)) {
+        return false;
     }
+
+    return true;
+}
+
+char *RobotArm::stradd(char *dest, const char *src) {
+    size_t i = 0, j = 0;
+
+    for (i = 0; dest[i] != '\0'; i++)
+        ;
```

Comment:
Please format like this:
for (;;) {
}

Reply:
This approach would not be favourable here, as in this case we will need to use ```break``` to check if the ```\0``` character is reached.

Reply:
I've misunderstand @vasilishomegas. I will change the for loop to the other format (bracket notation, instead of semicolon notation)